pkg/controller/job/job_controller_test.go

```diff
@@ -2265,6 +2266,12 @@ func TestSyncJobPastDeadline(t *testing.T) {
 			startTime:             10,
 			backoffLimit:          6,
 			expectedConditions: []batch.JobCondition{
+				{
+					Type:    batch.JobFailureTarget,
+					Status:  v1.ConditionTrue,
+					Reason:  batch.JobReasonDeadlineExceeded,
+					Message: "Job was active longer than specified deadline",
```

Comment:
We don't have consts for reasons, right?

Might make sense to introduce them as this messages are repeated in a couple of places across Job controller (in a followup PR though)

Reply:
I would only add a const if we use it in production code more than once.
If it's just to avoid repeating it in test cases, I'm not too in favor of it.

Reply:
I agree, I don't think the const for message is needed, but even if we conclude it is useful, I wouldn't change the testing approach in this PR.